[Fix] Use full name to look for the promoted prim operator table

iseeyuan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66052
* __->__ #66081

Since the operators are always registered with both name and overload name, the overloaded name need to be included when looking for an operator.

Differential Revision: [D31382262](https://our.internmc.facebook.com/intern/diff/D31382262/)

iseeyuan committed:
[Fix] Use full name to look for the promoted prim operator table

Since the operators are always registered with both name and overload name, the overloaded name need to be included when looking for an operator.

Differential Revision: [D31382262](https://our.internmc.facebook.com/intern/diff/D31382262/)

[ghstack-poisoned]

pavithranrao reviewed (approved):
Thanks for catching this, learnt few ops are overloaded.

iseeyuan committed:
Update on "[Fix] Use full name to look for the promoted prim operator table"


Since the operators are always registered with both name and overload name, the overloaded name need to be included when looking for an operator.

Differential Revision: [D31382262](https://our.internmc.facebook.com/intern/diff/D31382262/)

[ghstack-poisoned]

iseeyuan committed:
Update on "[Fix] Use full name to look for the promoted prim operator table"


Since the operators are always registered with both name and overload name, the overloaded name need to be included when looking for an operator.

Differential Revision: [D31382262](https://our.internmc.facebook.com/intern/diff/D31382262/)

[ghstack-poisoned]
